build.rs

```diff
@@ -12,22 +12,47 @@ fn main() {
 #[cfg(feature = "build_bindings")]
 mod generate {
     use std::error::Error;
+    use std::ffi::OsString;
     use std::io::Write;
     use std::process::{Command, Stdio};
 
     use regex::Regex;
     use std::env;
     use std::fs::File;
-    use std::path::Path;
+    use std::path::{Path, PathBuf};
 
     static CONST_PREFIX: &'static str = "DRM_FOURCC_";
 
+    pub fn get_header_include_paths() -> Vec<PathBuf> {
+        let library = pkg_config::Config::new()
+            .probe("libdrm")
+            .expect("Failed to find libdrm");
+
+        library.include_paths
+    }
+
     pub fn generate() -> Result<(), Box<dyn Error + Sync + Send>> {
+        let extra_include = get_header_include_paths();
         let out_dir = env::var("OUT_DIR").unwrap();
         let wrapper_path = Path::new(&out_dir).join("wrapper.h");
 
         // First get all the macros in drm_fourcc.h
 
+        let mut extra_include_paths = OsString::new();
+        {
+            let len = extra_include.len();
+            for i in 0..len {
+                extra_include_paths.push(&extra_include[i]);
+
+                if i != len {
+                    extra_include_paths.push(":");
+                }
+            }
+        }
+
+        // Tell clang where the header include paths are
+        env::set_var("C_INCLUDE_PATH", extra_include_paths);
```

Comment:
Can't you chain this through `.arg("-i").arg(extra_include)`, in a `for extra_include  in extra_includes`?

Reply:
bindgen further down the buildscript does not allow me to specify additional include paths apparently.

Reply:
oh wait `-i`.

Reply:
Okay I have used `-I`